### PR TITLE
Add Direct rounds support

### PR DESCRIPTION
**Pull request by lmcorbalan**

### CODE REVIEW

**Comment by bard** [comment]
@lmcorbalan please ignore the failed check — it's a bug in our workflow.

**Comment by bard** [comment]
@lmcorbalan just to confirm since I don't see reviews requested, should this be reviewed?

File: src/indexer/handleEvent.ts [hunk 1]
```diff
@@ -363,23 +368,45 @@ async function handleEvent(
...
+        const statusString = ApplicationStatus[status] as Application["status"];
         const application = await db
           .collection<Application>(`rounds/${event.address}/applications`)
-          .updateById(i.toString(), (application) => ({
-            ...application,
-            status: statusString as Application["status"],
-            statusUpdatedAtBlock: event.blockNumber,
-          }));
+          .updateById(i.toString(), (application) => {
+            const newApplication = {...application}
```
Comment (bard) [inline src/indexer/handleEvent.ts:375]: To help the type checker catch issues before the `return` statement, please declare the type for `newApplication`, as in:

```ts
const newApplication: Application =  {....application}
```

File: src/indexer/handleEvent.ts [hunk 2]
```diff
@@ -614,6 +641,76 @@ async function handleEvent(
...
+              assert: { type: "json" },
+            }
+          )
+        ).default,
+        event.blockNumber
+      );
+      break;
+    }
+
+    case "ApplicationInReviewUpdated": {
```
Comment (bard) [inline src/indexer/handleEvent.ts:661]: Is there any chance of factoring out common code between the `ApplicationInReviewUpdated` handler and the `ApplicationStatusesUpdated` handler?

**Comment by bard** [comment]
@lmcorbalan sorry for moving at snail pace with this. Just a quick check: was https://github.com/gitcoinco/statuses-bitmap/issues/3 in any way related to this?

**Comment by lmcorbalan** [comment]
> @lmcorbalan sorry for moving at snail pace with this. Just a quick check: was [gitcoinco/statuses-bitmap#3](https://github.com/gitcoinco/statuses-bitmap/issues/3) in any way related to this?

@bard Yes it was but I closed it because @gravityblast has a better fix for the issue https://github.com/gitcoinco/statuses-bitmap/pull/5

**Review by bard — APPROVED** [review]

**Pull request merged**